assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -179,6 +224,31 @@ function useMediaUploadQueue() {
               trackTiming();
             }
 
+            if (newResource.id) {
+              try {
+                const { poster, posterId } = await uploadVideoPoster(
+                  newResource.id,
+                  fileName,
+                  posterFile
+                );
+                const newResourceWithPoster = {
+                  ...newResource,
+                  poster: poster || newResource.poster || resource.poster,
+                  posterId,
+                };
+
+                finishUploading({
+                  id,
+                  resource: newResourceWithPoster,
+                });
+              } catch (error) {
+                finishUploading({
+                  id,
+                  resource: newResource,
+                });
```

Comment:
@swissspidy Now if poster upload fails for some reason, it will finish upload with a poster.

Reply:
Oh that's interesting!

I was not sure what to do in this case, but this makes total sense. Backfilling will then pick this up & try adding a poster again, which is fine.